kani-driver/src/project.rs

```diff
@@ -152,8 +149,8 @@ pub fn cargo_project(session: &KaniSession) -> Result<Project> {
     let outputs = session.cargo_build()?;
     let mut artifacts = vec![];
     let outdir = outputs.outdir.canonicalize()?;
-    if session.args.legacy_linker || session.args.function.is_some() {
-        // For the legacy linker or `--function` support, we still use a glob to link everything.
+    if session.args.function.is_some() {
+        // For the `--function` support, we still use a glob to link everything.
```

Comment:
Hmm, do we we need to? We can choose semantics for this, since it's not customer-facing. The only problem is trying to find the right target that has that symbol, right? Maybe we can just look it up! (Doesn't need to be this Pr, but open issue?)

Reply:
Totally. I think it's a separate issue, but I was thinking that we should just record the public functions as part of the metadata, like we do for tests. We could even restrict them to only include functions that don't take any parameter for now to avoid UB while using --function.

Reply:
https://github.com/model-checking/kani/issues/2129